specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/gallery.json

```diff
@@ -0,0 +1,2836 @@
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/GalleryProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/UpdateResourceDefinition"
+        }
+      ],
+      "description": "Specifies information about the Shared Image Gallery that you want to update."
+    },
+    "GalleryProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The description of this Shared Image Gallery resource. This property is updatable."
+        },
+        "identifier": {
+          "$ref": "#/definitions/GalleryIdentifier"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "title": "The current state of the gallery.",
+          "description": "The provisioning state, which only appears in the response.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Failed",
+            "Succeeded",
+            "Deleting",
+            "Migrating"
+          ]
+        },
+        "sharingProfile": {
+          "$ref": "#/definitions/SharingProfile"
+        }
+      },
+      "description": "Describes the properties of a Shared Image Gallery."
+    },
+    "GalleryIdentifier": {
+      "properties": {
+        "uniqueName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique name of the Shared Image Gallery. This name is generated automatically by Azure."
+        }
+      },
+      "description": "Describes the gallery unique name."
+    },
+    "SharingProfile": {
+      "description": "Profile for gallery sharing to subscription or tenant",
+      "properties": {
+        "permissions": {
+          "type": "string",
+          "description": "This property allows you to specify the permission of sharing gallery. <br><br> Possible values are: <br><br> **Private** <br><br> **Groups**",
+          "enum": [
+            "Private",
+            "Groups"
+          ],
+          "x-ms-enum": {
+            "name": "GallerySharingPermissionTypes",
+            "modelAsString": false
+          }
+        },
+        "groups": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharingProfileGroup"
+          },
+          "description": "A list of sharing profile groups."
+        }
+      }
+    },
+    "SharingProfileGroup": {
+      "description": "Group of the gallery sharing profile",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "This property allows you to specify the type of sharing group. <br><br> Possible values are: <br><br> **Subscriptions** <br><br> **AADTenants**",
+          "enum": [
+            "Subscriptions",
+            "AADTenants"
+          ],
+          "x-ms-enum": {
+            "name": "SharingProfileGroupTypes",
+            "modelAsString": false
```

Comment:
modelAsString=false is rare, why does this need to be a pure enum in generated sdks?  Same with the enum above this

Reply:
I see my team used a lot of this in the previous prs... Like OsType, OsState, SupportedOsType. In our api model, they are indeed enum type, not string.

Reply:
This doesn't have to do with your internal representation.  Setting modelAsString=false is telling the generated SDKs to use pure enums instead of strings.  It it still sent over the wire as a string according to your examples so why not model it in the SDK as a string to prevent breaking changes in the future if you reorder the enum, remove a value, etc..?  I imagine this was just a mistake made in the past

Reply:
Okay, I pushed new iteration which changes gallery sharing-related enum to be "modelAsString=true", since we might add new values in the same api version later.